docs/demos/filters/median_filter.jl

```diff
@@ -0,0 +1,83 @@
+# ---
+# title: Median filters
+# cover: assets/median.gif
+# author: Johnny Chen
+# date: 2020-09-23
+# ---
+
+# With `median` filter as an example, this demo shows how you could construct a custom kernal and
+# pass it to [`mapwindow`](@ref) for common stencil operations.
+
+# `ImageFiltering` does not provide you an out-of-box median filter (e.g., `medfilt2` in MATLAB).
+# This is because with [`mapwindow`](@ref) function, it is quite trivial a process to build and tailor one
+# such for your own usage.
+
+using ImageFiltering, ImageCore, ImageShow # or you could just `using Images`
+using TestImages
+using Statistics
+
+img = float.(reshape(1:25, 5, 5))
```

Comment:
Would it be better to use a testcase that's more obvious? The only places where the image changes in this example is along the edges, and you therefore have to know something about the default boundary conditions to make sense of the result.

What about something like
```julia
julia> img = [isodd(i) - isodd(j) for i = 1:5, j = 1:5]
5×5 Matrix{Int64}:
  0  1   0  1   0
 -1  0  -1  0  -1
  0  1   0  1   0
 -1  0  -1  0  -1
  0  1   0  1   0

julia> img[3, 3] = 1000
1000

julia> imgm = mapwindow(median, img, patch_size)
5×5 Matrix{Float64}:
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  1.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0
 0.0  0.0  1.0  0.0  0.0
 0.0  0.0  0.0  0.0  0.0
```